holobot/discord/sdk/workflows/interactables/component.py

```diff
@@ -11,8 +11,5 @@ class Component(Interactable):
     The same identifier CANNOT be used by multiple components to avoid ambiguity.
     """
 
-    component_type: type[ComponentBase]
+    component_type: type[ComponentBase] = field(repr=False)
     """The type of the component."""
-
-    def __str__(self) -> str:
-        return f"Component({self.identifier})"
```

Comment:
Leaving this note for myself not to forget about it.

I'll revert this change based on what we discussed:
* having the specific `__repr__` instead of manually excluding every other field makes more sense
* `__repr__` should ideally represent the entire object (kind of as a form of serialization) so that it can be restored at any time
* `__str__` is better in this case, because I want to see the original small amount of information versus the entire object representation in the logs

Reply:
Can I keep `__repr__` inheritance?
```py
@dataclass(kw_only=True)
class Component(Interactable):
    identifier: str
    component_type: type[ComponentBase]

    def __str__(self) -> str:
        return f"Component({self.identifier})"

@dataclass(kw_only=True, repr=False) # repr=False disables auto repr generation and inherits it from Component
class ModerationComponent(Component):
    required_moderator_permissions: ModeratorPermission = ModeratorPermission.NONE
```

Reply:
Reached consensus with `__str__` inheritance.
```py
@dataclass(kw_only=True)
class Component(Interactable):
    identifier: str
    component_type: type[ComponentBase]

    def __str__(self) -> str:
        return f"{type(self).__name__}({self.identifier})"

@dataclass(kw_only=True)
class ModerationComponent(Component):
    required_moderator_permissions: ModeratorPermission = ModeratorPermission.NONE
```